crates/octez/src/async/client.rs

```diff
@@ -331,6 +332,83 @@ impl OctezClient {
         self.spawn_and_wait_command(args).await?;
         Ok(())
     }
+
+    pub async fn originate_contract(
+        &self,
+        contract_alias: &str,
+        funding_account_src: &str, // can be an address or an alias
+        fund_tez: f64,
+        contract_path: &Path,
+        init_data: Option<&str>,
+        burn_cap: Option<f64>,
+    ) -> Result<(ContractKt1Hash, OperationHash)> {
+        let fund_str = fund_tez.to_string();
+        let burn_cap_str = match burn_cap {
+            Some(v) => v.to_string(),
+            None => String::new(),
+        };
+        let mut args = vec![
+            "originate",
+            "contract",
+            contract_alias,
+            "transferring",
+            &fund_str,
+            "from",
+            funding_account_src,
+            "running",
+            contract_path
+                .to_str()
+                .ok_or(anyhow::anyhow!("failed to convert contract path to string"))?,
+        ];
+        if let Some(data) = init_data {
+            args.append(&mut vec!["--init", data]);
+        }
+        if !burn_cap_str.is_empty() {
+            args.append(&mut vec!["--burn-cap", &burn_cap_str]);
+        }
+        let output = self.spawn_and_wait_command(args).await?;
+
+        let operation_hash = parse_operation_hash(&output);
+        if operation_hash.is_none() {
+            anyhow::bail!("failed to parse operation hash from execution output");
+        }
+        let contract_address = parse_contract_address(&output);
+        if contract_address.is_none() {
+            anyhow::bail!("failed to parse contract address from execution output");
+        }
+        Ok((contract_address.unwrap(), operation_hash.unwrap()))
+    }
+}
+
+fn parse_regex(pattern_str: &str, output: &str) -> Option<String> {
+    let pattern = regex::Regex::new(pattern_str).unwrap();
+    if let Some(capture) = pattern.captures(output) {
+        return Some(capture.get(1).unwrap().as_str().to_owned());
+    }
+    None
+}
+
+fn parse_operation_hash(output: &str) -> Option<OperationHash> {
+    if let Some(raw_operation_hash) =
+        parse_regex("Operation hash is '(o[1-9A-HJ-NP-Za-km-z]{50})'", output)
+    {
+        if let Ok(v) = OperationHash::from_base58_check(&raw_operation_hash) {
+            return Some(v);
+        }
+    }
+    None
+}
+
+fn parse_contract_address(output: &str) -> Option<ContractKt1Hash> {
+    if let Some(raw_contract_hash) = parse_regex(
+        "New contract (KT1[1-9A-HJ-NP-Za-km-z]{33}) originated.",
+        output,
+    ) {
+        if let Ok(v) = ContractKt1Hash::from_base58_check(&raw_contract_hash) {
+            return Some(v);
+        }
+    }
+    None
```

Comment:
```suggestion
    let raw_operation_hash =
        parse_regex("Operation hash is '(o[1-9A-HJ-NP-Za-km-z]{50})'", output)?;
    OperationHash::from_base58_check(&raw_operation_hash).ok()
```

Reply:
Updated. Sorry I'm not sure why I kept the awkward `if let` like that